ui.apps/test/components/commerce/navigation/navigationTest.js

```diff
@@ -0,0 +1,312 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+'use strict';
+
+import Navigation from '../../../../src/main/content/jcr_root/apps/core/cif/components/structure/navigation/v1/navigation/clientlibs/js/navigation';
+
+describe('Navigation', () => {
+    let FIRST_ITEM_SELECTOR = '.header__primaryActions aside div.categoryTree__root li:nth-child(1)';
+    var body;
+    var navigationRoot;
+
+    before(() => {
+        window.CIF = {
+            PageContext: {
+                maskPage: function() {},
+                unmaskPage: function() {}
+            }
+        };
+        body = window.document.querySelector('body');
+        navigationRoot = document.createElement('div');
+        body.insertAdjacentElement('afterbegin', navigationRoot);
+        navigationRoot.insertAdjacentHTML(
+            'afterbegin',
+            `
+                <div class="header__primaryActions">
+                    <button class="navTrigger__root clickable__root" aria-label="Toggle navigation panel">
+                        <span class="icon__root">
```

Comment:
This is only a "test" HTML so I think you can trim it down a little bit. One example - you don't need the full SVG here. As a general rule, we only need the elements defined by a selector so we can simulate clicks.